### PR TITLE
build: more informative publish-to-npm error message

### PR DESCRIPTION
**Pull request by ckerr**
#### Description of Change

We recently hit this error and the error message left some useful information out. This PR adds the url and status to the error message.

CC @VerteDinde @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Review by VerteDinde — APPROVED** [review]

File: script/release/publish-to-npm.js [hunk 1]
```diff
@@ -92,15 +92,16 @@ new Promise((resolve, reject) => {
...
-        url: tsdAsset.url,
+        url,
         headers: {
           accept: 'application/octet-stream',
           'user-agent': 'electron-npm-publisher'
         }
       }, (err, response, body) => {
         if (err || response.statusCode !== 200) {
-          reject(err || new Error('Cannot download electron.d.ts'));
+          reject(err || new Error(`Cannot download ${url}: ${response.status}`));
```
Comment (MarshallOfSound) [inline script/release/publish-to-npm.js:104]: We should also `console.error` the `body` and `response.headers`

Comment (MarshallOfSound) [reply script/release/publish-to-npm.js:108]: @ckerr Pretty sure `response.status` is `undefined`, you want `statusCode` here 🤔

Comment (ckerr) [reply script/release/publish-to-npm.js:108]: Added ff80c12a1833d599148c6a230c069febe783ab64

**Review by codebytere — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Comment by zcbenz** [comment]
@ckerr Can you rebase this PR?

**Comment by codebytere** [comment]
I think this can be closed in light of https://github.com/electron/electron/pull/30492 since it now logs all this information: https://github.com/electron/electron/blob/0c1f762119620913c7165b0002e93d11db11bf3b/script/release/get-asset.js#L28

Please reopen if i'm wrong though!

**Pull request closed without merging**